pkg/statdb/statdb.go

```diff
@@ -178,6 +178,7 @@ func (s *Server) Update(ctx context.Context, updateReq *pb.UpdateRequest) (resp
 
 // UpdateBatch for updating  multiple farmers' stats in the db
 func (s *Server) UpdateBatch(ctx context.Context, updateBatchReq *pb.UpdateBatchRequest) (resp *pb.UpdateBatchResponse, err error) {
+	// todo(moby) how should we handle one node failing to update but not all?
```

Comment:
code in here is looping in a for loop (line#186-198) while looping if the first node errors the kicks out of the loop completely. May be you should just log it and continue looping to update. After the for loop is done then handle the error by retying to update back the entry in to the DB.

Reply:
and if I retry the update what should I do if it fails again? When should I return an error and what should that error look like?

Reply:
Also, wouldn't it make more sense to fix this as a separate ticket? To fix it in this PR, I would be adding complexity that is not directly related to adding statdb to the satellite

Reply:
yep .. for now can u just log the err, just so there is a trail that we are bleeding... :-)